bootloader/ubootenv/env.go

```diff
@@ -58,18 +56,30 @@ func writeUint32(u uint32) []byte {
 	return buf.Bytes()
 }
 
+const sizeOfUint32 = 4
+
+func calcHeaderSize(headerFlagByte bool) int {
+	if headerFlagByte {
+		// If uboot uses a header flag byte, header is 4 byte crc + flag byte
+		return sizeOfUint32 + 1
+	}
+	// otherwise, just a 4 byte crc
+	return sizeOfUint32
+}
+
 // Create a new empty uboot env file with the given size
-func Create(fname string, size int) (*Env, error) {
+func Create(fname string, size int, headerFlagByte bool) (*Env, error) {
```

Comment:
I would prefer to have something more self-descriptive here instead of a bool argument. The issue is that it's unclear from the calling code what the bool means, e.g. when reading:
```go
env, err := ubootenv.Create(u.envFile, 4096, true)
``` 
One has to lookup the function docstring and it breaks reading the code. Minor issue given how rarely this is used but something like:
```go
type CreateOpts struct { RedudnantEnvironment bool }
func Create(fname string, size, CreateOpts *opts) (*Env, error) {
```
or 
```go
const RedundantEnv = iota + 1
func Create(fname string, size int, flags  int) (*Env, error) {
```
With that the call side feels easier to read:
```go
env, err := ubootenv.Create(u.envFile, 4096, ubootenv.RedudantEnv)
env, err := ubootenv.Create(u.envFile, 4096, ubootenv.CreateOpts{RedudantEnv: true})
```
no strong opinion on which one to use though.

Reply:
Thanks for the feedback! I see what you mean there. I've updated to use the struct version. I used a struct member name that was more consistent with what the naming I used elsewhere in the PR, but I am not really attached to it one way or the other and am happy to convert it (and other variables) to something more `RedundEnv`ish instead of `HeaderFlagByte` (I went with the latter because I guess theoretically uboot could update to use the upper 7 bits of the byte to signify something. Seems unlikely, but technically possible).